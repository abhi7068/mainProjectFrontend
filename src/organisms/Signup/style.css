.change {
  color: lightblue;
  text-decoration: underline;
}
.change:hover {
  color: red;
}
