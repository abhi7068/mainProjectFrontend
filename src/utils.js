import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
