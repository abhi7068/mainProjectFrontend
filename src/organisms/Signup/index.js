import React, { useState } from 'react';
import { Formik, Form } from 'formik';
import * as Yup from 'yup';
import FormikField from '../../molecules/FormikField';
import { signUpApi } from '../../services/signUpServices';
import Button from '@material-ui/core/Button';
import { Link } from 'react-router-dom';
import { Alert, AlertTitle } from '@material-ui/lab';
import LinkComponent from '../../atoms/Link';
import './style.css';

const initialValues = {
  name: '',
  email: '',
  confirmEmail: '',
  password: '',
  confirmPassword: '',
};

const SignupSchema = Yup.object().shape({
  name: Yup.string(
    'Enter full name as alphabets only, no numbers and special characters should be given'
  )
    .min(2, 'Name must contain at least 2 characters')
    .max(30, 'Name must contain no more then 20 characters')
    .required('Name is required'),
  email: Yup.string('Enter your email')
    .email('Enter a valid email')
    .required('Email is required')
    .max(35, 'Email must contain no more then 35 characters')
    .matches(
      /^[A-Za-z].*$/,
      'Email should not start with number orspecial character'
    ),
  confirmEmail: Yup.string('Enter your email')
    .required('Confirm your email')
    .oneOf([Yup.ref('email')], 'Email does not match')
    .max(35, 'Email must contain no more then 35 characters'),
  password: Yup.string('Enter a Password')
    .min(8, 'Password must contain at least 8 characters')
    .max(20, 'Password must contain no more then 20 characters')
    .required('Enter your password')
    .matches(
      /(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[\x21-\x2F\x3A-\x40\x5B-\x60\x7B-\x7E]).{8,20}/,
      // @#$%|!*>_<={+/,.";:'&()}?^~`-
      'Password must contain at least one uppercase,one number, one special character'
    )
    .matches(/^[A-Z]/, 'Password must start withupper case'),
  confirmPassword: Yup.string('Enter a password')
    .required('Confirm your password')
    .oneOf([Yup.ref('password')], 'Enter correct password'),
});

const Signup = () => {
  const [message, setMessage] = useState({ error: '', success: '' });
  const handleSubmit = (values, { resetForm }) => {
    signUpApi(values).then((response) => {
      if (response && response.status === 200) {
        setMessage({ success: response.data.message });
        resetForm({});
        // props.history.push('/login');
      } else setMessage({ error: 'CANNOT SIGNUP' });
    });
  };

  return (
    <div style={{ paddingBottom: '2em' }}>
      {message && message.error ? (
        <Alert severity="error">
          <AlertTitle>Error</AlertTitle>
          {message.error}
        </Alert>
      ) : (
        ''
      )}

      {message && message.success ? (
        <Alert severity="success">
          <AlertTitle>Success</AlertTitle>
          {message.success}
        </Alert>
      ) : (
        ''
      )}
      <Formik
        initialValues={initialValues}
        onSubmit={handleSubmit}
        validationSchema={SignupSchema}
      >
        {({ dirty, isValid, resetForm }) => {
          return (
            <Form>
              <FormikField name="name" label="Name" required />
              <FormikField name="email" label="Email" required />
              <FormikField name="confirmEmail" label="Confirm email" required />
              <FormikField
                name="password"
                type="password"
                label="Password"
                required
              />
              <FormikField
                name="confirmPassword"
                type="password"
                label="Confirm Password"
                required
              />
              <br />
              <br />
              <Button
                variant="contained"
                color="primary"
                disabled={!dirty || !isValid}
                type="submit"
              >
                Signup
              </Button>
              &nbsp;
              <Button
                data-testid="Reset"
                variant="contained"
                type="reset"
                color="primary"
                onClick={resetForm}
              >
                Reset
              </Button>
              <br />
              <br />
              <LinkComponent>
                Already have an account?&#160;
                <Link to="/login">
                  <span className="change">Login</span>
                </Link>
              </LinkComponent>
              <br />
            </Form>
          );
        }}
      </Formik>
    </div>
  );
};
export default Signup;
