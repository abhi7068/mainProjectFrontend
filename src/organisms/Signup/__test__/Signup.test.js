import React from 'react';
import ReactDOM from 'react-dom';
import Signup from '../index';
import { render, fireEvent } from '../../../utils';
import { Provider } from 'react-redux';
import reducer from '../../../store/reducers/rootReducer';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';
const store = createStore(reducer, applyMiddleware(thunk));
it('it renders', () => {
  render(<Signup />);
});

it('should take a snapshot', () => {
  const { asFragment } = render(<Signup />);
  expect(asFragment(render(<Signup />))).toMatchSnapshot();
});
it('renders with a expected length', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <BrowserRouter>
        <Signup />
      </BrowserRouter>
    </Provider>,
    div
  );
  expect(div.querySelectorAll('input')).toHaveLength(5);
});
it('renders with a expected attribute', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <BrowserRouter>
        <Signup />
      </BrowserRouter>
    </Provider>,
    div
  );
  expect(div.querySelector('input')).toHaveAttribute('type', 'text');
});
it('to be true', () => {
  const { getByTestId } = render(<Signup />);
  expect(fireEvent.click(getByTestId('Reset'))).toBe(true);
});
