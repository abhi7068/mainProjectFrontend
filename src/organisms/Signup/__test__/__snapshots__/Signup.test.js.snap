// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    style="padding-bottom: 2em;"
  >
    <form
      action="#"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
        >
          Name
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiInput-input"
            name="name"
            required=""
            type="text"
            value=""
          />
        </div>
        <p
          class="MuiFormHelperText-root Mui-required"
        />
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
        >
          Email
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiInput-input"
            name="email"
            required=""
            type="text"
            value=""
          />
        </div>
        <p
          class="MuiFormHelperText-root Mui-required"
        />
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
        >
          Confirm email
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiInput-input"
            name="confirmEmail"
            required=""
            type="text"
            value=""
          />
        </div>
        <p
          class="MuiFormHelperText-root Mui-required"
        />
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
        >
          Password
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiInput-input"
            name="password"
            required=""
            type="password"
            value=""
          />
        </div>
        <p
          class="MuiFormHelperText-root Mui-required"
        />
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
        >
          Confirm Password
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiInput-input"
            name="confirmPassword"
            required=""
            type="password"
            value=""
          />
        </div>
        <p
          class="MuiFormHelperText-root Mui-required"
        />
      </div>
      <br />
      <br />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Signup
        </span>
      </button>
       
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        data-testid="Reset"
        tabindex="0"
        type="reset"
      >
        <span
          class="MuiButton-label"
        >
          Reset
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <br />
      <br />
      <p
        class="MuiTypography-root MuiTypography-body2"
        data-testid="hello"
        mx="20"
      >
        Already have an account? 
        <a
          href="/login"
        >
          <span
            class="change"
          >
            Login
          </span>
        </a>
      </p>
      <br />
    </form>
  </div>
</DocumentFragment>
`;
